src/main/java/org/testng/TestClass.java

```diff
@@ -26,6 +28,14 @@
   private XmlClass xmlClass;
   private final String m_errorMsgPrefix;
 
+  @Override
+  public Map<Object, ITestNGMethod[]> getBeforeClassConfig() {
+    return beforeClassConfig;
+  }
+
+  private Map<Object, ITestNGMethod[]> beforeClassConfig = new HashMap<>();
```

Comment:
@bj-9527 - Since the `Object` is just being used as the key, we can instead just rely on the `hashCode()` combination to determine which object to query. I believe that the `Map` construct always uses the `hashCode()` to determine the relevant bucket and the `equals()` to further narrow down the absolute match. 
Since this is being internally used by TestNG, ca't we not change this to just capture the `hashCode()` as the key instead of the entire Test Class object ?

Also as @juherr mentioned, this can be moved into a separate interface which only `TestClass` will implement and we can internally do a `instance of ` check to determine whether we should be using this or not.